### PR TITLE
Implement Puppeteer architecture & add basic test

### PR DESCRIPTION
**Pull request by juliaamosova**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

- Implement initial Puppeteer's architecture;
- Remove visual regression testing library implementation for now (will re-enable it when we are done with E2E test suite migration from Selenium to Puppeteer and write new tests);
- Add a basic e2e test to make sure Puppeteer's architecture is working;
- Add README for setting up and running E2E tests. 

### How to test the changes in this Pull Request:

1. Setup local E2E test environment as described in the new README file of this PR;
2. Run E2E test headless by using `npm run test:e2e`;
3. Run E2E test in `Development mode` (non-headless with SlowMo enabled). `Development mode` can be enabled by passing `--dev` flag to the `test:e2e script` in `./package.json` file as follows:

`"test:e2e": "./tests/bin/e2e-test-integration.js --dev",`

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### Changelog entry

> Implement Puppeteer's architecture for running E2E tests locally.

### CODE REVIEW

**Comment by juliaamosova** [comment]
Pinging @woocommerce/proton for review.

File: package.json [hunk 1]
```diff
@@ -16,21 +16,24 @@
     "test": "cross-env NODE_CONFIG_DIR='./tests/e2e-tests/config' BABEL_ENV=commonjs mocha \"tests/e2e-tests\" --require babel-register --recursive",
```
Comment (kloon) [inline package.json]: This and the two other below can be removed since it references the old e2e tests?

File: package.json [hunk 2]
```diff
@@ -16,21 +16,24 @@
...
+    "@babel/cli": "^7.5.5",
+    "@babel/core": "^7.5.5",
+    "@babel/polyfill": "^7.4.4",
+    "@babel/preset-env": "^7.5.5",
+    "@babel/register": "^7.5.5",
     "babel-eslint": "10.0.2",
-    "babel-plugin-add-module-exports": "1.0.2",
-    "babel-preset-es2015": "6.24.1",
-    "babel-preset-stage-2": "6.24.1",
+    "babel-jest": "^24.9.0",
```
Comment (kloon) [inline package.json]: We have babel-jest and jest at line 63 of this file. Not sure what the difference is, but perhaps only one is needed?

**Review by kloon — CHANGES_REQUESTED** [review]
Codewise this is looking good to me, just a small couple of questions then it should be fine to approve from my side.

File: package.json (hunk 1 above)
Comment (juliaamosova) [reply package.json]: @kloon you are right. I removed the legacy e2e test scripts.

File: package.json (hunk 2 above)
Comment (juliaamosova) [reply package.json]: @kloon good catch. Thanks for a detailed review. According to [here](https://github.com/babel/babel-jest), `babel-jest` is now automatically loaded by `Jest` and fully integrated. I removed `babel-jest`. Tested without it - everything is working.

**Review by kloon — APPROVED** [review]
Thanks for changes, good to merge!

**Pull request merged**